addons/stock/models/stock_move.py

```diff
@@ -1465,6 +1506,33 @@ def unlink(self):
         self.with_context(prefetch_fields=False).mapped('move_line_ids').unlink()
         return super(StockMove, self).unlink()
 
+    def _get_consuming_document(self):
+        """ TODO WIP """
+        self.ensure_one()
+        return self.picking_id or False
+
+    def _get_replenishment_document(self):
+        """ TODO WIP """
+        self.ensure_one()
+        return self.picking_id or False
+
+    def _is_consuming(self, wh_location_ids):
+        """ TODO WIP """
+        self.ensure_one()
+        if wh_location_ids and self.picking_type_id.code == 'internal':
+            return self.location_id.id in wh_location_ids
```

Comment:
It's not True for me. Maybe the move could remains in an internal location. I would add a check on destination location (if it's in customer or not under the same warehouse).